steward/src/somm_send.rs

```diff
@@ -86,16 +86,16 @@ pub async fn data_hash(
     val_address: String,
 ) -> Result<AllocationPrecommit, Error> {
     let mut hasher = sha2::Sha256::new();
-    if let Some(cellar) = &allocation.clone().vote.unwrap().cellar {
+    if let Some(vote) = &allocation.clone().vote {
         let mut buf = BytesMut::new();
-        cellar.encode(&mut buf).unwrap();
-        let cellar_data = hex::encode(&buf).to_string();
-        let msg = format!("{}:{}:{}", allocation.salt, cellar_data, val_address);
+        vote.encode(&mut buf).unwrap();
+        let vote_data = hex::encode(&buf).to_string();
+        let msg = format!("{}:{}:{}", allocation.salt, vote_data, val_address);
```

Comment:
I think I found part of the cause of the hash mismatch problem but it still doesnt match. It looks like somm is encoding then hashing the entirety of the RebalanceVote whereas steward was just doing the Cellar part of the vote object. Could you confirm I made the right fix here @EricBolten ? Compare with the following:

https://github.com/PeggyJV/sommelier/blob/4cc93e29f506ada4a572d845646ad3b3353a44dd/x/allocation/types/rebalance_vote.go#L28-L39

https://github.com/PeggyJV/sommelier/blob/4cc93e29f506ada4a572d845646ad3b3353a44dd/x/allocation/types/allocation.pb.go#L556-L564

Reply:
Yes, it appears that the chain side is encoding the entire `RebalanceVote` and here we were only using the cellar field.

Reply:
It is technically a race but it works in my experience because the next vote period is ~15-20 seconds away when the rebalance request gets sent (at least on my machine). I feel like that is enough room to call it safe.. you?